torchmetrics/metric.py

```diff
@@ -341,16 +416,17 @@ def persistent(self, mode: bool = False):
     def state_dict(self, destination=None, prefix="", keep_vars=False):
         destination = super().state_dict(destination=destination, prefix=prefix, keep_vars=keep_vars)
         # Register metric states to be part of the state_dict
-        for key in self._defaults:
-            if self._persistent[key]:
-                current_val = getattr(self, key)
-                if not keep_vars:
-                    if torch.is_tensor(current_val):
-                        current_val = current_val.detach()
-                    elif isinstance(current_val, list):
-                        current_val = [cur_v.detach() if torch.is_tensor(cur_v) else cur_v for cur_v in current_val]
-                destination[prefix + key] = current_val
-        return destination
+        with self.sync_context(dist_sync_fn=self.dist_sync_fn):
+            for key in self._defaults:
+                if self._persistent[key]:
+                    current_val = getattr(self, key)
+                    if not keep_vars:
+                        if torch.is_tensor(current_val):
+                            current_val = current_val.detach()
+                        elif isinstance(current_val, list):
+                            current_val = [cur_v.detach() if torch.is_tensor(cur_v) else cur_v for cur_v in current_val]
+                    destination[prefix + key] = deepcopy(current_val)
```

Comment:
Why do we need the deepcopy now?

Reply:
As the state dict is being synced across processes, and we don't want to keep the reference.

Reply:
```suggestion
                    # the tensors will be synced across processes so deepcopy to drop the references
                    destination[prefix + key] = deepcopy(current_val)
```